cdci_data_analysis/analysis/instrument.py

```diff
@@ -539,48 +553,67 @@ def set_input_products(self, par_dic, input_file_path,input_prod_list_name):
 
                 par_dic[input_prod_list_name] = cleaned_lines
                 has_prods= len(lines) >= 1
-            except:
+            except RequestNotUnderstood as e:
+                raise e
+            except Exception:
                 has_prods=False
 
         return has_prods
 
     def upload_catalog_from_fronted(self, par_dic, request, temp_dir):
+        cat_file_path = None
         if request.method == 'POST':
             # save to a temporary folder, and delete it afterwards
             cat_file_path = upload_file('user_catalog_file', temp_dir)
             if cat_file_path is not None:
                 par_dic['user_catalog_file'] = cat_file_path
+        return cat_file_path
 
     def set_catalog(self, par_dic):
-        user_catalog_file = None
-        if 'user_catalog_file' in par_dic.keys():
-            user_catalog_file = par_dic['user_catalog_file']
-
-        if 'user_catalog_dictionary' in par_dic.keys() and par_dic['user_catalog_dictionary'] is not None:
-            if type(par_dic['user_catalog_dictionary']) == dict:
-                self.set_par('user_catalog', build_catalog(par_dic['user_catalog_dictionary']))
-            else:
-                catalog_dic = json.loads(par_dic['selected_catalog'])
-                self.set_par('user_catalog', build_catalog(catalog_dic))
         # setting user_catalog in the par_dic, either loading it from the file or aas an object
-        if user_catalog_file is not None:
-            catalog_object = load_user_catalog(user_catalog_file)
+        if 'user_catalog_file' in par_dic.keys() and par_dic['user_catalog_file'] is not None:
+            user_catalog_file = par_dic['user_catalog_file']
+            try:
+                catalog_object = load_user_catalog(user_catalog_file)
+            except RuntimeError:
+                raise RequestNotUnderstood('format not valid, a catalog should be provided as a FITS (typical standard OSA catalog) or '
+                                           '<a href=https://docs.astropy.org/en/stable/api/astropy.io.ascii.Ecsv.html>ECSV</a> table.')
             self.set_par('user_catalog', catalog_object)
-            # TODO set selected_catalog ? in this way it will show up in the email url link
-            # and used consistently for the job_id generation
             self.set_par('selected_catalog', json.dumps(catalog_object.get_dictionary()))
-            # TODO verify that the file path is not needed in the frontend (should not)
+            # TODO not needed in the frontend
             par_dic.pop('user_catalog_file', None)
         else:
+            # TODO is this case still in use?
```

Comment:
Is this still used for catalogs?

Reply:
I think that's what passed in some cases when frontend send the table. Well, you can verify if you want.

Reply:
Just checked, and it is indeed passed from the frontend!

Reply:
this is tested now